src/coreclr/jit/morph.cpp

```diff
@@ -13446,6 +13441,26 @@ GenTree* Compiler::fgOptimizeEqualityComparisonWithConst(GenTreeOp* cmp)
     {
         ssize_t op2Value = static_cast<ssize_t>(op2->IntegralValue());
 
+        // Optimizes (X & 1) == 1 to (X & 1)
+        //
+        //                        EQ/NE                   AND
+        //                        /  \                   /  \.
+        //                      AND   CNS 1  ->         x   CNS 1
+        //                     /   \.
+        //                    x   CNS 1
+        //
+        // The compiler requires jumps to have relop operands, so we do not fold that case.
+        if (op1->OperIs(GT_AND) && op2Value == 1 && !(cmp->gtFlags & GTF_RELOP_JMP_USED))
```

Comment:
```suggestion
        if (op1->OperIs(GT_AND) && (op2Value == 1) && !(cmp->gtFlags & GTF_RELOP_JMP_USED))
```
[See the code guide](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-jit-coding-conventions.md#111-logical-and-arithmetic-expressions).